WeakAuras/WeakAuras.lua

```diff
@@ -5733,3 +5787,55 @@ function WeakAuras.SafeToNumber(input)
   local nr = tonumber(input)
   return nr and (nr < 2147483648 and nr > -2147483649) and nr or nil
 end
+
+function WeakAuras.SetHistory(uid, data, fromAddon, addon)
+  if uid and data then
+    db.history[uid] = db.history[uid] or {}
+    db.history[uid].data = data
+    db.history[uid].source = fromAddon and "addon" or "import"
+    db.history[uid].addon = fromAddon and (addon or "unknown") or nil
+    db.history[uid].skippedVersions = db.history[uid].skippedVersions or {}
+    db.history[uid].lastUpdate = time()
+  end
+end
+
+function WeakAuras.GetHistory(uid)
+  return uid and db.history[uid]
+end
+
+function WeakAuras.RemoveHistory(uid)
+  if uid then
+    db.history[uid] = nil
+  end
+end
+
+function WeakAuras.SkipVersion(uid, version, skip)
+  local history = WeakAuras.GetHistory(uid)
+  if history then
+    history.skippedVersions[version] = skip
+  end
+end
+
+function WeakAuras.IsVersionSkipped(uid, version)
+  local history = WeakAuras.GetHistory(uid)
+  if history then
+    return history.skippedVersions[version]
+  end
+end
+
+function WeakAuras.RestoreFromHistory(uid)
```

Comment:
This can be used to undelete, and revert changes. I believe that it will be useful to make this restoration more intelligent about parent/child relationships eventually, but I have not yet decided how that should happen.